guides/rif-relay/smart-wallets.md

```diff
@@ -16,135 +21,46 @@ There are **two ways** to create a Smart Wallet:
-      import Web3 from "web3";
-
-      const web3 = new Web3(<RSK_NODE_ENDPOINT>);
-
-      const token = await new this.web3.eth.Contract(
-        <ERC20_TOKEN_ABI>,
-        <TOKEN_ADDRESS>
-      );
-     const accounts = await this.web3.eth.getAccounts();
-     await token.methods.transfer(<SMART_WALLET_ADDRESS>, this.web3.utils.toWei(<AMOUNT_OF_TOKENS>, "ether"))
-          .send({ from: accounts[0] });
-   ```
-   Where variables are:
 
-   * **RSK_NODE_ENDPOINT**: the RSK node endpoint where is running (ex: http://localhost:4444).
-   * **ERC20_TOKEN_ABI**: the ERC20 token contract abi to use.
-   * **TOKEN_ADDRESS**: the token contract address.
-   * **SMART_WALLET_ADDRESS**: the generated address of last step.
-   * **AMOUNT_OF_TOKENS**: string containing the amount of tokens in decimal unit.
-
-   **NOTE: in this example we asume that the `account[0]` of regtest has tokens to use.**
-
-3. Finally you need to deploy your smart wallet, to do so you must follow these steps:
-
-   1. Use the provider from relay to that you have an example here:
-   ```javascript
-
-      import { RelayProvider, resolveConfiguration } from "@rsksmart/rif-relay";
-      import Web3 from "web3";
-
-      const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
-
-      const web3 = new Web3(<RSK_NODE_ENDPOINT>);
-
-      const config = await resolveConfiguration(web3.currentProvider,
-      {
-        verbose: window.location.href.includes("verbose"),
-        onlyPreferredRelays: true,
-        preferredRelays: ["http://localhost:8090"],
-        factory: <SMART_WALLET_FACTORY_CONTRACT_ADDRESS>,
-        gasPriceFactorPercent: 0,
-        relayLookupWindowBlocks: 1e5,
-        chainId: 33,
-        relayVerifierAddress: <RELAY_VERIFIER_CONTRACT_ADDRESS>,
-        deployVerifierAddress: <DEPLOY_VERIFIER_CONTRACT_ADDRESS>,
-        smartWalletFactoryAddress: <SMART_WALLET_FACTORY_CONTRACT_ADDRESS>
-      });
-      resolvedConfig.relayHubAddress = <RELAY_HUB_CONTRACT_ADDRESS>;
-
-      const provider = new RelayProvider(web3.currentProvider, config);
-
-      provider.addAccount({
-         address: <RSK_ACCOUNT_ADDRESS>,
-         privateKey: Buffer.from(<RSK_ACCOUNT_PRIVATE_KEY>.replaceAll("0x", ""), "hex")
-      });
-
-      web3.setProvider(provider);
-
-      const transaction = await provider.deploySmartWallet({
-        from: <RSK_ACCOUNT_ADDRESS>,
-        to: ZERO_ADDRESS,
-        gas: "0x27100",
-        value: "0",
-        callVerifier: <DEPLOY_VERIFIER_CONTRACT_ADDRESS>,
-        callForwarder: <SMART_WALLET_FACTORY_CONTRACT_ADDRESS>,
-        tokenContract: <TOKEN_ADDRESS>,
-        tokenAmount: <TOKEN_AMOUNT>,
-        data: "0x",
-        index: <SMART_WALLET_INDEX>,
-        recoverer: <SMART_WALLET_RECOVERER>,
-        isSmartWalletDeploy: true,
-        onlyPreferredRelays: true,
-        smartWalletAddress: <SMART_WALLET_ADDRESS>,
-      });
+At this point we should have the SDK object created and being called from a dApp. 
```

Comment:
Should we include here both the SDK and client code (e.g.: using a (basic/low level approach))?
Do you think there are things can be specified only using the client?

Reply:
I think that there could be multiple approach, using the client since, we need to include the way that the requests are build and explain about each field. We could include this, that will provide a better understanding from the flow.